src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp

```diff
@@ -96,8 +96,10 @@ QVariant QgsValueRelationWidgetWrapper::value() const
       }
     }
 
-    if ( layer()->fields().at( fieldIdx() ).type() == QVariant::Map )
+    if ( layer()->fields().at( fieldIdx() ).type() == QVariant::Map ||
+         layer()->fields().at( fieldIdx() ).type() == QVariant::List )
     {
+      //because of json it's stored as QVariantList with type QVariant::Map or QVariant::List
```

Comment:
Maybe JSON, maybe some other way for structured data in the data provider. I think that's the wrong level to talk about storage types.

Reply:
Okay, i'll remove these comments... In the end it's rather confusing than helpful.